storage/database_test.go

```diff
@@ -0,0 +1,471 @@
+package storage
+
+import (
+	"bytes"
+	"crypto/rand"
+	"github.com/golang/glog"
+	"github.com/stretchr/testify/require"
+	"math"
+	"os"
+	"testing"
+)
+
+type TestConfig struct {
+	// BatchSizeBytes is the size of the batch in bytes.
+	BatchSizeBytes int
+	// BatchSizeItems is the number of items in the batch.
+	BatchSizeItems int
+	// BatchItemsRemoved is the number of items removed from the batch as part of the experiment.
+	BatchItemsRemoved int
+	// BatchItemsRetrieved is the number of items retrieved from the batch as part of the experiment.
+	BatchItemsRetrieved int
+	// BatchItemsIterated is the number of items iterated over in the batch as part of the experiment.
+	BatchItemsIterated int
+}
+
+type Key [32]byte
+
+func NewKey(key []byte) Key {
+	var k Key
+	copy(k[:], key[:])
+	return k
+}
+
+func (k Key) Bytes() []byte {
+	copyKey := make([]byte, len(k[:]))
+	copy(copyKey, k[:])
+	return copyKey
+}
+
+type KeyValue struct {
+	Key   Key
+	Value []byte
+}
+
+func NewKeyValue(key []byte, value []byte) *KeyValue {
+	return &KeyValue{
+		Key:   NewKey(key),
+		Value: value,
+	}
+}
+
+// RandomBytes returns a []byte with random values.
+func RandomBytes(numBytes int32) []byte {
+	randomBytes := make([]byte, numBytes)
+	_, err := rand.Read(randomBytes)
+	if err != nil {
+		glog.Errorf("Problem reading random bytes: %v", err)
+	}
+	return randomBytes
+}
+
+// TestBadger_Default_Experiment_10MB_Batch is a BadgerDB test in which we write 10MB of data to the database.
+// In the test, we:
+//  1. Write 100 equal size KV items to the database.
+//  2. Remove 20 items from the database.
+//  3. Retrieve 20 items from the database.
+//  4. Iterate over 20 items in the database.
+//  5. Iterate over all items in the database.
+func TestBadger_Default_Experiment_10MB_Batch(t *testing.T) {
```

Comment:
This is totally on your radar, but Im dropping the comment here so it's contextualized in the PR too. We only need one test using Default options and one test using the Performance options. And in general, both tests need to write the minimum amount of data to disk to test the logic you've built in the wrapper structs above. 

These test will execute in every CI run, so we want to avoid writing large amounts of data to disk and to instead use the minimum overhead needed to test the new logic.

Reply:
also note that we should update `test.Dockerfile` to run tests in this directory by adding `"github.com/deso-protocol/core/storage"` on line 32